### PR TITLE
Upload to s3 using httpotion

### PR DESCRIPTION
**Pull request by shankardevy**
Tested on my local machine. It seems to be uploading without error.

Note:
this uses [Calendar](https://github.com/lau/calendar) instead of Timex. Calendar seems to be nicely integrating with Ecto and have good apis. There are about 3 calls to Timex in the code which are easy to replace with Calendar and I provide a separate patch for it. Timex used in this project is pretty old and has bugs relating to timezone conversion when used with dateformat. Newer version may have resolved this but contains many breaking changes between 0.13-0.16.

### CODE REVIEW

**Comment by mosic** [comment]
@shankardevy Uploading seems to succeed without errors, but the uploaded file contains only the filename and not the actual image. The problem lies in `body: tmp_path` part.

That confused me for a minute :smile:

**Comment by shankardevy** [comment]
@mosic good catch. rectified. Also verified in S3 and the images are viewable.

File: lib/media/s3.ex [hunk 1]
```diff
@@ -13,26 +13,30 @@ defmodule EvercamMedia.S3 do
...
+    string = "PUT\n\n#{content_type}\n#{date}\n/#{System.get_env("AWS_BUCKET")}#{file_path}"
+    signature = :crypto.hmac(:sha, System.get_env("AWS_SECRET_KEY"), string) |> Base.encode64
+    authorization = "AWS #{System.get_env("AWS_ACCESS_KEY")}:#{signature}"
 
-    if output.err || output.status != 0 do
-      raise HTTPotion.HTTPError, message: inspect(output.err)
+    headers = ["Host": host,
+               "Date": date,
+               "Content-Type": content_type,
+               "Authorization": authorization]
```
Comment (mosic) [inline lib/media/s3.ex]: Please update this to be in the format of:

```
headers = [
  "Host": host,
  "Date": date,
  "Content-Type": content_type,
  "Authorization": authorization
]
```

It's not that important, but just for the sake of consistency I'd prefer that indentation style.

**Comment by mosic** [comment]
@shankardevy Great.

It seems that you haven't signed the [CLA](https://www.clahub.com/agreements/evercam/evercam-media) yet, let me know when you do and I'll merge this.

**Comment by shankardevy** [comment]
@mosic I agree we need a consistent code style in our repos. I have updated the code and agreed to CLA.

**Comment by mosic** [comment]
@shankardevy Great, thanks.

**Comment by shankardevy** [comment]
Thank you :-)

**Pull request merged**